BootloaderCorePkg/Stage2/Stage2.c

```diff
@@ -527,6 +528,16 @@ SecStartup (
     }
   }
 
+  //
+  // Allocate SMBIOS tables' memory, set Base and call Smbios init
+  //
+  SmbiosEntry = AllocateZeroPool (PcdGet16(PcdSmbiosTablesSize));
+  PcdSet32S (PcdSmbiosTablesBase, (UINT32)(UINTN)SmbiosEntry);
+  Status = SmbiosInit ();
+  if (EFI_ERROR(Status)) {
```

Comment:
Is this mandatory for all platforms? Not an optional?

Reply:
Could you please point out any concerns here if it is added by default for all platforms?

Reply:
There is no concern, but I think better to give an option to enable/disable SmBiosTable depending on Platform. As Guo mentioned, the SmBiosInitLib can be part of BootloaderCommonPkg instead of BootloaderCorePkg.

Reply:
Placed the init code inside a Pcd check to see if platform enables SMBIOS. Default is disabled.